### PR TITLE
[AzureClient] Expose feature gates for AzureClient users

### PR DESCRIPTION
**Pull request by scottn12**
## Description

This PR exposes feature gates for AzureClient users to allow the use of experimental features.

Please see [Design Doc (MSFT Internal)](https://microsoft.sharepoint-df.com/:fl:/g/contentstorage/CSP_5fcca107-c97c-4b0a-899e-42a2578d1609/EfTkHbkax_hBuFfYisduTSMBSx_kKMQ_kmascMi0pYzT2Q?e=HKPjXg&nav=cz0lMkZjb250ZW50c3RvcmFnZSUyRkNTUF81ZmNjYTEwNy1jOTdjLTRiMGEtODk5ZS00MmEyNTc4ZDE2MDkmZD1iJTIxQjZITVgzekpDa3VKbmtLaVY0MFdDY3JoRTNub2dqNURqZ01yWHlBTGQ3MWxiZ0VwY3VvYlFKXzhnb2luWjJFTyZmPTAxTUdUSktNUFU0UU8zU0dXSDdCQTNRVjZZUkxEVzRUSkQmYz0lMkYmYT1Mb29wQXBwJnA9JTQwZmx1aWR4JTJGbG9vcC1wYWdlLWNvbnRhaW5lciZ4PSU3QiUyMnclMjIlM0ElMjJUMFJUVUh4dGFXTnliM052Wm5RdWMyaGhjbVZ3YjJsdWRDMWtaaTVqYjIxOFlpRkNOa2hOV0RONlNrTnJkVXB1YTB0cFZqUXdWME5qY21oRk0yNXZaMm8xUkdwblRYSlllVUZNWkRjeGJHSm5SWEJqZFc5aVVVcGZPR2R2YVc1YU1rVlBmREF4VFVkVVNrdE5UMFZDTWxGRFRVSlZSMVpDUVV4R1RGVkpNelpOTlVaT1JqTSUzRCUyMiUyQyUyMmklMjIlM0ElMjJjN2M4MmY5Ni0xMTRhLTQ2MDAtODFlYy1jMzc5YTM1ZDZjZmMlMjIlN0Q%3D) for context.

## Reviewer Guidance
I would appreciate feedback on:
- Using `ExperimentalFlags` instead of `IConfigProviderBase` (see design doc "Alternative Approaches" section for more info)
- Implementation details
- Where should supporting documentation go?

## Misc
[AB#3946](https://dev.azure.com/fluidframework/235294da-091d-4c29-84fc-cdfc3d90890b/_workitems/edit/3946)

### CODE REVIEW

File: api-report/azure-client.api.md [hunk 1]
```diff
@@ -45,6 +46,7 @@ export class AzureClient {
 // @public
 export interface AzureClientProps {
     readonly connection: AzureRemoteConnectionConfig | AzureLocalConnectionConfig;
+    readonly experimentalFlags?: ExperimentalFlags;
```
[the remark this answers is not in the document]
Comment (scottn12) [reply api-report/azure-client.api.md:49]: I was talking about this with @ssimic2, and the thought was that if there was no explicit use case that we shouldn't give AzureClient users more options than they would need. I am definitely open to changing this if we feel it is valuable/something we will want in the long run
cc @anthony-murphy @ssimic2 @markfields

[the remark this answers is not in the document]
Comment (anthony-murphy) [reply api-report/azure-client.api.md:49]: I just worry about divergence between the two stacks. this change introduces completely new concepts that don't map to existing concepts (experimental, flags). It also creates a new and unique abstraction layer. I don't think doing this for every feature is sustainable long term, and we should aim to minimize divergence where possible.

**Comment by scottn12** [comment]
> @scottn12 Mind if I add a changeset to this PR before you merge?

Please do @tylerbutler, thanks!

File: api-report/azure-client.api.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (anthony-murphy) [reply api-report/azure-client.api.md]: if you think the existing abstraction could be improved that could also be done, which would then bubble up here.

**Review by anthony-murphy — APPROVED** [review]

File: azure/packages/azure-client/src/AzureClient.ts [hunk 2]
```diff
@@ -21,6 +21,7 @@ import { IClient, SummaryType } from "@fluidframework/protocol-definitions";
 import { RouterliciousDocumentServiceFactory } from "@fluidframework/routerlicious-driver";
 import { requestFluidObject } from "@fluidframework/runtime-utils";
 
+import { IConfigProviderBase } from "@fluidframework/telemetry-utils";
```
Comment (anthony-murphy) [inline azure/packages/azure-client/src/AzureClient.ts:24]: maybe file an issue for this, but i think we should consider moving this interface to somewhere more easily shared, like container-definitions

**Pull request merged**